test/unit/parachain/asset-registry.test.ts

```diff
@@ -0,0 +1,141 @@
+import { expect } from "../../chai";
+import sinon from "sinon";
+import { ApiPromise } from "@polkadot/api";
+import { StorageKey, u32 } from "@polkadot/types";
+import { OrmlAssetRegistryAssetMetadata } from "@polkadot/types/lookup";
+import { DefaultAssetRegistryAPI } from "../../../src/";
+import { AssetRegistryMetadataTuple } from "@interlay/interbtc/parachain/asset-registry";
+
+describe("DefaultAssetRegistryAPI", () => {
+    let api: ApiPromise;
+    let assetRegistryApi: DefaultAssetRegistryAPI;
+    let mockMetadata: OrmlAssetRegistryAssetMetadata;
+    const mockMetadataValues =  {
+        name: "Mock Coin One",
+        symbol: "MCO",
+        decimals: 8,
+        existentialDeposit: 42,
+        feesPerMinute: 15
+    };
+
+    before(() => {
+        api = new ApiPromise();
+        // disconnect immediately to avoid printing errors
+        // we only need the instance to create variables
+        api.disconnect();
+
+        // register just enough from OrmlAssetRegistryAssetMetadata to construct 
+        // meaningful representations for our tests
+        api.registerTypes({
+            OrmlAssetRegistryAssetMetadata: {
+                name: "Bytes",
+                symbol: "Bytes",
+                decimals: "u32",
+                existentialDeposit: "u128",
+                additional: "InterbtcPrimitivesCustomMetadata"
+            },
+            InterbtcPrimitivesCustomMetadata: {
+                feePerSecond: "u128"
+            }
```

Comment:
Awesome, `registerTypes` is just what we needed. As a future todo, we should try to fetch the decoded metadata from the running chain and build the expected type dynamically, so the unit tests do break when the parachain is upgraded. Example for `OrmlAssetRegistryAssetMetadata` (the hardcoded `50` array index should be replaced but stil): 
```bash
$ subsee --format json --url http://localhost:9933 | jq ".[1] | .V14.types.types | .[50].type.def"
```

Reply:
I have a feeling that parts of the polkadotjs typegen could also be reused for this, worth checking their tests